unified_frameworks/sensor_array/server_lidar.py

```diff
@@ -17,57 +17,78 @@ def getDevicePort():
 
 port = getDevicePort()
 
-print(f"Port = {getDevicePort()}")
+# @note Quick encoder from numpy's ndarray to json
+# @note since json cannot serialize ndarrays
+class NumpyEncoder(json.JSONEncoder):
+
+    def default(self,obj):
+        if isinstance(obj, np.ndarray):
+            return obj.tolist()
+
+# lidar = RPLidar(port)
+lidar = None
 
 if port is None:
-    print("Port not found!")
-    sys.exit(1)
+    print("[SERVER] Device port not found!")
+    print("[SERVER] Using Fake Lidar")
+    lidar = FakeLidar()
+else:
+    print(f"[SERVER] Device port {port} found!")
+    lidar = ActualLidar()
 
-lidar = RPLidar(port)
-clients = [] # List of clients connected in the server
+clients = [] # @note List of clients connected in the server
 buffer = None
 
 # Creating WebSocket server
-async def serverConnections(websocket):
-    global lidar
-
+# @note Server listens for ConnectionClosedOK
+# @note Sends data from the server to the client via websockets
+# @note checks if the lidar we are sending data from is either FakeLidar or ActualLidar
+# @note If current lidar is FakeLidar then we utilize NumpyEncoder.
+async def sendServerDataToClient(websocket):
     clients.append(websocket)
+    print("[SERVER] Client has connected to the server")
+    # await websocket.send(json.dumps("[SERVER] You have connected to the server!"))
+    # await asyncio.sleep(0.01)
+    isConnectedStill = True
 
     try:
-        while True:
-            # Receiving values from client (lidar)
-            # RAW DATA is in tuples
-            for scan in lidar.iter_scans():
-                buffer = json.dumps(scan)
+        while isConnectedStill:
+            buffer = None
+
+            # @note Checking lidars instance to see which data the buffer should contain
+            # @note Buffer will send the raw data from the lidar to the client
+            if isinstance(lidar, FakeLidar):
+                buffer = json.dumps(lidar.get_measures(), cls=NumpyEncoder)
```

Comment:
Dude, just make FakeLidar return its result as a list instead of converting the ndarray into a list over here.